src/backend/distributed/planner/multi_logical_planner.c

```diff
@@ -738,6 +739,22 @@ WhereClauseList(FromExpr *fromExpr)
 
 
 /*
+ * OuterJoinClauseList walks over the FROM expression in the query tree, and builds
+ * a list of outer join clauses from the expression tree.
+ */
+List *
+OuterJoinClauseList(FromExpr *fromExpr)
```

Comment:
We have used a struct a pass multiple parameters to walker function. One example is [here](https://github.com/citusdata/citus/blob/master/src/backend/distributed/planner/multi_logical_optimizer.c#L1229) in ```multi_logical_optimizer.c```